src/jsonschema/converter.py

```diff
@@ -0,0 +1,73 @@
+#  Copyright (c) 2022 spdx contributors
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from abc import ABC, abstractmethod
+from typing import Any, Type, Dict, TypeVar, Generic
+
+from src.jsonschema.json_property import JsonProperty
+from src.model.document import Document
+from src.writer.casing_tools import snake_case_to_camel_case
+
+MISSING_IMPLEMENTATION_MESSAGE = "Must be implemented"
+
+T = TypeVar("T")
+
+
+class TypedConverter(ABC, Generic[T]):
+    """
+    Base class for all converters between an instance of the tools-python data model and the corresponding dictionary
+    representation, following the json schema. The generic type T is the type according to the data model.
+    Each converter has several methods:
+    - get_json_type and get_data_model_type: return the data model type and the corresponding JsonProperty subclass.
+    These methods are abstract in the base class and need to be implemented in subclasses.
+    - json_property_name: converts an enum value of a JsonProperty subclass to the corresponding property name in the
+    json schema. The default implementation simply converts to snake case. Can be overridden in case of exceptions
```

Comment:
```suggestion
    json schema. The default implementation simply converts from snake case to camel case. Can be overridden in case of exceptions
```